tests/test.rs

```diff
@@ -532,8 +563,8 @@ fn gnu_apple_sysroot() {
         test.shim("fake-gcc")
             .gcc()
             .compiler("fake-gcc")
-            .target(&target)
-            .host(&target)
+            .target(target)
+            .host(target)
             .file("foo.c")
             .compile("foo");
 
```

Comment:
If we _actually_ want to support shared libraries, we should consider a `.linker` and `.get_linker` similar to `.archiver` and `.get_archiver`, as well as maybe allowing overriding the linker with an `LD` env var (?)

Though if we allow using a raw linker (e.g. invoke `rust-lld` binary instead of `cc`), we're getting into hairy territory of needing to know how to invoke the linker if it's not a compiler driver.

Reply:
(Which again is why I'm really not in favour of it).